Amplify/Categories/API/Reachability/ReachabilityUpdate.swift

```diff
@@ -0,0 +1,15 @@
+//
+// Copyright 2018-2019 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+public struct ReachabilityUpdate {
```

Comment:
Does this gain us anything beyond a simple Bool?

Reply:
For now, you are right, it's just a bool, but since it is crossing the boundary between two categories, I believe that introducing a type that we own/control allows us to make it easier to make changes to the interface.  In this case, we may want to reveal additional information like the connection type (wifi, cell, microwave.. hah, etc.)

Reply:
I'm not convinced; in order to make this an additive change, we'd have to retain the `isOnline` boolean property and add new specifiers for connection type, which would then be redundant with `isOnline`, and we don't yet have a "different connection type" story. 

If this ends up being a public API, I'll push back harder on it during the API review, but for now we can leave it as-is for an internal type just to prove functionality.